extmod/network_lwip.c

```diff
@@ -90,6 +99,279 @@ mp_obj_t mod_network_nic_ifconfig(struct netif *netif, size_t n_args, const mp_o
+                            while (!dhcp_supplied_address(netif)) {
+                                if (mp_hal_ticks_ms() - start > 10000) {
+                                    mp_raise_msg(&mp_type_OSError, MP_ERROR_TEXT("timeout waiting for DHCP to get IP address"));
+                                }
+                                mp_hal_delay_ms(100);
+                            }
+                        } else {
+                            dhcp_release_and_stop(netif);
+                        }
+                        break;
+                    }
+                    #if LWIP_IPV6_DHCP6
+                    case MP_QSTR_dhcp6: {
+                        dhcp6_disable(netif);
+                        dhcp6_enable_stateless(netif);
+                        break;
+                    }
+                    #endif
+                    #if LWIP_IPV6_AUTOCONFIG
+                    case MP_QSTR_autoconf6: {
+                        netif_set_ip6_autoconfig_enabled(netif, mp_obj_is_true(e->value));
+                        if (mp_obj_is_true(e->value)) {
+                            nd6_restart_netif(netif);
+                            uint32_t start = mp_hal_ticks_ms();
+                            while (1) {
+                                int found = 0;
+                                for (i = 1; i < LWIP_IPV6_NUM_ADDRESSES; i++) {
+                                    if (ip6_addr_isvalid(netif_ip6_addr_state(netif, i)) &&
+                                        !netif_ip6_addr_isstatic(netif, i)) {
+                                        found = 1;
+                                        break;
+                                    }
+                                }
+                                if (found) {
+                                    break;
+                                }
+                                if (mp_hal_ticks_ms() - start > 10000) {
+                                    mp_raise_msg(&mp_type_OSError, MP_ERROR_TEXT("timeout waiting for DHCP to get IP address"));
+                                }
+                                mp_hal_delay_ms(100);
+                            }
+                        } else {
+                            // Clear out any non-static addresses, skip link-local address in slot 0
+                            for (i = 1; i < LWIP_IPV6_NUM_ADDRESSES; i++) {
+                                if (ip6_addr_isvalid(netif_ip6_addr_state(netif, i)) &&
+                                    !netif_ip6_addr_isstatic(netif, i)) {
+                                    netif_ip6_addr_set_state(netif, i, IP6_ADDR_INVALID);
+                                }
+                            }
+                        }
+                        break;
+                    }
+                    #endif
+                    case MP_QSTR_addr4:
+                    case MP_QSTR_addr6: {
+                        ip_addr_t ip_addr;
+                        int prefix_bits = 32;
```

Comment:
Following up to the earlier discussion about `"ip/N"` vs `("ip", "mask")`, I think we should make this function accept both.

i.e. if `e->value` is str, then use what you've implemented below, else use `mp_obj_get_array_fixed_n` and extract the ip and mask.

The rationale is that the setter should accept the same type as the getter, and I think that the tuple makes sense for the getter.